### ISSUE TITLE
Add means to ensure project.version when not SNAPSHOT matches git tag on current commit 

### ISSUE THREAD

**Issue by roytmana** [ktoso/maven-git-commit-id-plugin#106]
When making a release build, it would be nice to ensure that pom version exactly matches tag  and terminate the build if it does not. If maven version is NOT a SNAPSHOT, we should ensure current commit has a label attached that exactly matches pom version. This way we can ensure the versions are in sync and we can safely use pom version.

We may want to check if SNAPSHOT version is greater first reachable tag as well but that's not that important (and not that simple because it requires observing maven version comparision rues)

**Comment by TheSnoozer** [comment]
Hi @roytmana ,
I'm currently thinking about the best way to implement this feature/improvement. My first draft looks like adding a List of validationProperties containing one or multiple Objects of a validationProperty. For each validationProperty we could add a name and a regex to match against. If any of the regexes does not match the name, we could control the failure of the build by the trigger ``failIfNoMatch``. The List option allows to match different thinks like git.dirty and git.commit.id at the same time.

Currently I'm not sure how we can evaluate whether we need to execute the check or not.
I think we'll end up going for another goal that will be executed in ``<phase>install</phase>`` (I have a project where created different goals that will be executed in different phases; you can configure them with annotations). However I think making it fixed to some phase (e.g. install) may not satisfy other user requirements (maybe they want to execute it on validate or whatsoever)

If we set the phase in the configuration of the git-commit-id-plugin to install the whole complete git information extracting and validation will be executed at install (kind of late if you need the properties in your project).

Adding another configuration with different execution phases of the git-commit-id-plugin to the pom seems like nothing won in comparison to the configuration of the build-helper-maven-plugin.

We definitely need to investigate if we somehow can inject the execution of this to a random phase.

```
<plugin>
	<groupId>pl.project13.maven</groupId>
	<artifactId>git-commit-id-plugin</artifactId>
	<version>2.1.14</version>
	<configuration>
		<validationProperties>
			<validationProperty>
				<value>${project.version}</value>
				<regex>^\Q${git.commit.id.describe}\E$|.*-SNAPSHOT$</regex>
			</validationProperty>
			<validationProperty>
				<value>${git.dirty}</value>
				<regex>false</regex>
			</validationProperty>
			<failIfNoMatch>true</failIfNoMatch>
		</validationProperties>
	</configuration>
</plugin>
```

**Comment by ktoso** [comment]
Impl looks good, I've merged #254
